### ISSUE TITLE
`torch.optim.Optimizer.zero_grad()` MLX equivalent

### ISSUE THREAD

**Issue by a1eaiactaest** [ml-explore/mlx#459]
looking for the [`zero_grad()`](https://pytorch.org/docs/stable/generated/torch.optim.Optimizer.zero_grad.html) equivalent in mlx. help appreciated.

**Comment by a1eaiactaest** [comment]
will zeroing out the gradients by hand work?

**Comment by awni** [comment]
There is no `zero_grad` in MLX. When you get the gradient of a function it returns a new variable which you can use as you like. So something like:

```
loss, grads = nn.value_and_grad(model, loss_fn)
```

Give you gradients that you can then use to update the model's parameters. If you do it again you get a different set of arrays.

**Comment by a1eaiactaest** [comment]
Thank you
